assets/src/design-system/components/button/index.js

```diff
@@ -109,36 +107,68 @@ const buttonColors = {
 
 const ButtonRectangle = styled(Base)`
   ${({ type }) => type && buttonColors?.[type]};
-  min-width: 80px;
-  height: 36px;
-  padding: 8px 16px;
+  min-width: 1px;
+  min-height: 1em;
   border-radius: 4px;
+
+  & > div {
+    display: flex;
+    align-items: center;
+    justify-content: space-around;
+    box-sizing: border-box;
+    padding: ${({ size }) =>
+      size === BUTTON_SIZES.SMALL ? '8px 16px' : '16px 32px'};
+    height: 100%;
+  }
 `;
 
-const ButtonCircle = styled(Base)`
+const ButtonSquare = styled(Base)`
   ${({ type }) => type && buttonColors?.[type]};
-
+  border-radius: 4px;
   ${({ size }) => css`
-    width: ${size === BUTTON_SIZES.SMALL ? 32 : 56}px;
-    height: ${size === BUTTON_SIZES.SMALL ? 32 : 56}px;
-    border-radius: 50%;
+    width: ${(size === BUTTON_SIZES.SMALL ? 32 : 56) + 8}px;
+    height: ${(size === BUTTON_SIZES.SMALL ? 32 : 56) + 8}px;
+
+    & > div {
+      display: flex;
+      align-items: center;
+      justify-content: space-around;
+      box-sizing: border-box;
+      padding: 10px 10px;
+      height: 100%;
+      width: 100%;
+    }
 
-    & > svg {
-      width: ${size === BUTTON_SIZES.SMALL ? 14 : 20}px;
-      height: ${size === BUTTON_SIZES.SMALL ? 14 : 20}px;
+    svg {
+      width: ${size === BUTTON_SIZES.SMALL ? 16 : 20}px;
+      height: ${size === BUTTON_SIZES.SMALL ? 16 : 20}px;
       margin: 0 auto;
     }
   `}
 `;
 
+const ButtonCircle = styled(ButtonSquare)`
+  border-radius: 50%;
+`;
+
 const ButtonIcon = styled(Base)`
   ${({ size }) => `
-    width: ${size === BUTTON_SIZES.SMALL ? 14 : 20}px;
-    height:${size === BUTTON_SIZES.SMALL ? 14 : 20}px;
-
-    & > svg {
+    width: ${(size === BUTTON_SIZES.SMALL ? 16 : 20) + 8}px;
+    height:${(size === BUTTON_SIZES.SMALL ? 16 : 20) + 8}px;
+
+    & > div {
+      display: flex;
+      align-items: center;
+      justify-content: space-around;
+      box-sizing: border-box;
+      height: 100%;
+      width: 100%;
+    }
+    
+    svg {
       width: 100%;
       height:100%;
```

Comment:
just realizing this, and know it's unrelated to the changes here, but should one of these be `auto` to preserve the aspect ratio of the svg?

Might be fine tho since it doesn't happen on the select cases we'd be using this 🤷‍♂️

Reply:
good point, i'll swap height to auto to be on the safe side. def will keep an eye out as i test out the buttons more aggressively in different components.